test/recipes/25-test_req.t

```diff
@@ -15,13 +15,24 @@ use OpenSSL::Test qw/:DEFAULT srctop_file/;
 
 setup("test_req");
 
-plan tests => 4;
+plan tests => 8;
 
 require_ok(srctop_file('test','recipes','tconversion.pl'));
 
 open RND, ">>", ".rnd";
 print RND "string to make the random number generator think it has randomness";
 close RND;
+
+# Check for duplicate -addext parameters
+my $val = "subjectAltName=DNS:example.com";
+my $val2 = " " . $val;
+my $val3 = $val;
+$val3 =~ s/=/    =/;
+ok(!run(app(["openssl", "req", "-new", "-addext", $val, "-addext", $val])));
+ok(!run(app(["openssl", "req", "-new", "-addext", $val, "-addext", $val2])));
+ok(!run(app(["openssl", "req", "-new", "-addext", $val, "-addext", $val3])));
+ok(!run(app(["openssl", "req", "-new", "-addext", $val2, "-addext", $val3])));
```

Comment:
-addext doesn't work at all now. I mean not even if you pass one. This is because lh_TYPE_insert returns NULL on success [and error].

On related note it should be noted that these tests don't actually prove anything. Indeed, imagine that req erroneously pass duplicates, what would happen? If executed in terminal test will hang asking for PEM pass phrase, i.e. ultimately fail.

Reply:
They are negative tests, they show that multiple copies fail.   There were no positive tests, so when I broke the working case the tests did not show it.  Thanks for fixing it.  These tests are still useful, as that is the original "feature" I was adding.

Reply:
Purpose of the tests is to **catch** things, right? In order to catch you need **distinguishable** outcomes. In this case application's exit code is used. Point is that it will always return same condition, error, and tests is bound to succeed no matter what. I mean it returns error with single -addext and with duplicate -addext options, and you can't tell failure and success apart. They should be formed in a way that if duplicate -addext is omitted, test would actually succeed.